mods/default/functions.lua

```diff
@@ -481,3 +481,35 @@ minetest.register_abm({
 		end
 	end
 })
+
+
+--
+-- Checks if specified volume intersects a protected volume
+--
+
+-- Checks 7 corners first then interior volume at specified intervals
+-- minp can be skipped as it is checked in the loop below
+
+function default.intersects_protection(minp, maxp, name, interval)
+	if minetest.is_protected({x = minp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = maxp.z}, name) then
+		return true
+	end
+
+	for z = minp.z, maxp.z, interval do
+	for y = minp.y, maxp.y, interval do
+	for x = minp.x, maxp.x, interval do
```

Comment:
How about code like the following (disclaimer: not tested) for avoiding re-checking the corners. Assuming a protection check can be expensive, and assuming the volume to check is not too large, this would be more efficient:
```
-- Avoid re-checking the corners
-- (assuming protection checks can be expensive,
--  and the checked volume is not too large.)
local corner = 0
corner = corner + ((x == minp.x or x == maxp.x) and 1 or 0)
corner = corner + ((y == minp.y or y == maxp.y) and 1 or 0)
corner = corner + ((z == minp.z or z == maxp.z) and 1 or 0)
if corner < 3 then
	-- do protection check
end
```